openmc/mgxs_library.py

```diff
@@ -0,0 +1,716 @@
+from collections import Iterable
+from numbers import Real, Integral
+from xml.etree import ElementTree as ET
+import warnings
+import sys
+if sys.version_info[0] >= 3:
+    basestring = str
+
+import numpy as np
+
+import openmc
+from openmc.mgxs import EnergyGroups
+from openmc.checkvalue import check_type, check_value, check_greater_than, \
+                              check_iterable_type
+from openmc.clean_xml import *
+
+# Supported incoming particle MGXS angular treatment representations
+REPRESENTATIONS = ['isotropic', 'angle']
```

Comment:
Roger, I actually was thinking it would end up being used by something else if we tie this to MGXSLibrary, but for now, private it up!